### ISSUE TITLE
Feature to cancel http request promise

### ISSUE THREAD

**Issue by dheran** [angular/angular#8525]
Provide an option to cancel an http request.

**Comment by dloukadakis** [comment]
U can already do that by calling **unsubscribe** on the object returned by subscribe method.

example:
```
var subscription = http.get(url)
      .map(res => res.json())
      .subscribe(data => this.data = data);

// this call will cancel the http request
subscription.unsubscribe();
```

**Comment by dheran** [comment]
how can we bring similar behavior to Promise

**Comment by zoechi** [comment]
One main difference between `Promise` and `Observable` is that `Promise` is not cancelable while `Observable` is. See also http://stackoverflow.com/a/29479435/217408 or http://www.bennadel.com/blog/2731-canceling-a-promise-in-angularjs.htm

**Comment by pkozlowski-opensource** [comment]
Thnx @zoechi !

**Comment by theMSY** [comment]
@dloukadakis The problem with that method is that the call persists in the network queue even after unsubbing .

**Comment by AckerApple** [comment]
I have learned a few things worth reporting. At first I was looking at this issue report like "YOU'RE ALL LIERS!" but I have figured myself out and it's worth noting:

- If you unsubscribe IMMEDIATELY after a request is created AND the request has a pre OPTIONS request, the OPTIONS request is cancelled and NOT the actual POST request. Rare to ever happen in production but was common during my testing
- I HAD WRONG IDEA: Each time you use the subscribe method it sends another HTTP request. I thought .subscribe() meant to listen to the events occurring but subscribe actually means "send and subscribe". SO, when I was cancelling using .unsubscribe() but had multiple .subscribe() calls I was a rabbit running down a hole of bad assumptions.

Hope this helps someone else
